### PR TITLE
Created infer tool

### PR DESCRIPTION
**Pull request by omartiny**
Created cli infer tool for conditional compilation based

### CODE REVIEW

File: tests/conditional_compilation/test_infer_tool.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import argparse
```
Comment (asomsiko) [inline tests/conditional_compilation/test_infer_tool.py:1]: The file name should not start with `test` - those names are reserved for test cases/suites. Please also have a separate folder for the tool

./tests/conditional_compilation/tools/infer_tool.py

File: tests/conditional_compilation/test_infer_tool.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import logging as log
+import sys
+
+log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+import os
```
Comment (asomsiko) [inline tests/conditional_compilation/test_infer_tool.py:6]: please ensure `pip3 install pylint` `pylint <file name>` report is clear

File: tests/conditional_compilation/test_infer_tool.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import numpy as np
+from openvino.inference_engine import IENetwork
+
+
+def python_api_infer(ir_path, device):
```
Comment (asomsiko) [inline tests/conditional_compilation/test_infer_tool.py:12]: What is the reason for `python_api` in a function name? Should be name it `infer` or `do_infer`?

File: tests/conditional_compilation/test_infer_tool.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    del net
+    # It's important to delete executable network first to avoid double free in plugin offloading.
+    # Issue relates ony for hetero and Myriad plugins
+    del exec_net
+    del ie
+    return res
+
+
+def cli_parser():
+    parser = argparse.ArgumentParser(description='Python_api reproducer')
```
Comment (asomsiko) [inline tests/conditional_compilation/test_infer_tool.py:38]: description needs to be rewritten.

**Review by asomsiko — CHANGES_REQUESTED** [review]

**Review by asomsiko — APPROVED** [review]

File: tests/conditional_compilation/tools/infer_tool.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+def infer(ir_path, device):
+    """
+     Function to perform IE inference using python API "in place"
+    :param ir_path: Path to XML file of IR
+    :param device: Device name for inference
+    :return: Dict containing out blob name and out data
+    """
+
+    bin_path = os.path.splitext(ir_path)[0] + '.bin'
+    net = IENetwork(model=ir_path, weights=bin_path)
```
Comment (vurusovs) [inline tests/conditional_compilation/tools/infer_tool.py:28]: please, use `ie.read_network` instead

File: tests/conditional_compilation/tools/infer_tool.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    feed_dict = {}
+    for layer, name in net.inputs.items():
+        feed_dict.update({layer: np.ones(shape=name.shape)})
```
Comment (vurusovs) [inline tests/conditional_compilation/tools/infer_tool.py:31]: please, create a separate function with input preparation. Function docstring should describe the aim that input should be reproducible from run to run

File: tests/conditional_compilation/tools/infer_tool.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def cli_parser():
+    """
+    Function for parsing arguments from command line.
+    :return: ir path, device and output folder path variables.
+    """
+    parser = argparse.ArgumentParser(description='Arguments for python API inference')
+    parser.add_argument('-m', dest='ir_path', required=True, help='Path to XML file of IR')
+    parser.add_argument('-d', dest='device', help='Target device to infer on', default="CPU")
```
Comment (vurusovs) [inline tests/conditional_compilation/tools/infer_tool.py:51]: please, make argument required with no default value

File: tests/conditional_compilation/tools/infer_tool.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    parser.add_argument('-r', dest='out_path', default=None,
```
Comment (vurusovs) [inline tests/conditional_compilation/tools/infer_tool.py:52]: please, make argument required with no default value

File: tests/conditional_compilation/tools/infer_tool.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                        help='Dumps results to the output folder')
```
Comment (vurusovs) [inline tests/conditional_compilation/tools/infer_tool.py:53]: Argument means full path to a file, not a folder. Please, fix help

File: tests/conditional_compilation/tools/infer_tool.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    args = parser.parse_args()
+    ir_path = args.ir_path
+    device = args.device
+    out_path = args.out_path
+    return ir_path, device, out_path
+
+
+if __name__ == "__main__":
+    ir_path, device, out_path = cli_parser()
+    bin_path = os.path.splitext(ir_path)[0] + '.bin'
```
Comment (vurusovs) [inline tests/conditional_compilation/tools/infer_tool.py:63]: unused var

**Review by vurusovs — COMMENTED** [review]
Please, create a separate PR with patch according to notes

**Pull request merged**